keps/sig-windows/2258-node-service-log-viewer/README.md

```diff
@@ -0,0 +1,395 @@
+      --raw=false: Perform no transformation of the returned data.
+      --role='': Set a label selector by node role.
+  -l, --selector='': Selector (label query) to filter on.
+      --since='': Return logs after a specific ISO timestamp or relative date. Only applies to node journal or Get-WinEvent logs.
+      --tail=0: Return up to this many lines (not more than 100k) from the end of the log. Only applies to node journal or Get-WinEvent logs.
+      --unify=false: Interleave logs by sorting the output. Defaults on when viewing node journal logs.
+  -s, --service=[]: Return log entries from the specified service(s). Only applies to node journal or Get-WinEvents logs.
+      --until='': Return logs before a specific ISO timestamp or relative date.
+```
+
+Given that the feature will be introduced behind a feature gate, by default
+`kubectl logs nodes` will return a feature not enabled message. When the
+feature is enabled in alpha phase, `kubectl logs nodes` will display a
+warning message that the feature is in alpha. When the `--service` option
+is used against Linux nodes that do not support systemd/journald, an OS
+not supported message will be returned.
+
+A reference implementation of this feature can be see in the OpenShift
+[oc adm node-logs](https://github.com/openshift/origin/pull/21701/commits/d9dc6892d395c239b19831641218ca047e74ad94)
+sub-command.
+
+### Test Plan
+Add unit tests to kubelet and kubectl that exercise the new arguments that
+have been added. A reference implementation of the tests can be seen
+[here](https://github.com/kubernetes/kubernetes/pull/96120/commits/c606a38ec38ccfe486033495a1dc433279ce71f8#diff-1d703a87c6d6156adf2d0785ec0174bb365855d4883f5758c05fda1fee8f7f1bR1)
+
+### Graduation Criteria
+
+The plan is to introduce the feature as alpha in the v1.21 time frame behind the
+`NodeLogs` feature gate.
+
+#### Alpha -> Beta Graduation
+
+The plan is to graduate the feature to beta in the v1.22 time frame. At that
+point we would have collected feedback from cluster administrators and
+developers who have enabled the feature. Based on this feedback and issues
+opened we should consider adding a kubelet side throttle for the viewing the
+logs.
+
+#### Beta -> GA Graduation
+
+The plan is to graduate the feature to GA in the v1.23 time frame at which point
+any major issues would have been surface during the alpha and beta phases.
+
+### Upgrade / Downgrade Strategy
+
+### Version Skew Strategy
+
+If a kubectl version that has the new `logs nodes --service` option is used
+against a node that is using a kubelet that does not have the extended
+`/var/log` endpoint viewer, the result should be "feature not supported".
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [x] Feature gate
+    - Feature gate name: NodeLogs
+    - Components depending on the feature gate: kubelet
```

Comment:
There is really no change needed to kube-apiserver to support this? Proxy will do this without any change?

Reply:
That is correct. @smarterclayton implemented this in OpenShift and we have it working. Please see the [kubelet](https://github.com/kubernetes/kubernetes/pull/96120) and corresponding [oc](https://github.com/openshift/origin/pull/21701/commits/d9dc6892d395c239b19831641218ca047e74ad94) changes for a POC.